### PR TITLE
Change barcode entry

### PR DESCRIPTION
**Pull request by apmasell**
This commit changes how barcodes are displayed:
- a print button in a toolbar for printing a barcode is on appropriate pages
- if barcodes are editable, the barcode is displayed as a normal field
- the datamatrix for the barocode is no longer displayed on the page

As side effects:
- no temporary files are created for barcodes
- barcode4j is no longer needed
- CHS code is reduce
- a DIV to display jQuery dialogs was moved to the header

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -84,10 +84,6 @@
       <artifactId>joda-time</artifactId>
     </dependency>
     <dependency>
-      <groupId>net.sf.barcode4j</groupId>
-      <artifactId>barcode4j</artifactId>
```
Comment (djcooke) [inline core/pom.xml]: Can be removed from parent pom as well

File: miso-web/src/main/webapp/scripts/hot.js [hunk 2]
```diff
@@ -803,7 +803,7 @@ var HotUtils = {
   
   printAction : function(type) {
     return {
-      name : 'Print',
+      name : 'Print Barcode',
```
Comment (djcooke) [inline miso-web/src/main/webapp/scripts/hot.js]: "Barcode(s)"

**Comment by callunity** [comment]
All of the edit single X pages have the new Matrix Barcode field right below the alias, but I think it makes more sense to place it just below the description (flip the matrix barcode and description rows). This keeps it from getting visually 'lost' when it's shorter than both the alias and the description, and is more in line with the ordering we have in the bulk tables. What do you think?

**Comment by callunity** [comment]
Could the `Dilutions` table on the Edit Library page be changed to include the matrix barcode? That's not currently user-visible anywhere other than the bulk create/edit pages.

**Comment by apmasell** [comment]
Rebased and fixed.

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/SampleControllerHelperService.java [hunk 3]
```diff
@@ -54,18 +47,13 @@
...
-import uk.ac.bbsrc.tgac.miso.core.data.Boxable;
 import uk.ac.bbsrc.tgac.miso.core.data.Sample;
 import uk.ac.bbsrc.tgac.miso.core.data.impl.ProjectOverview;
-import uk.ac.bbsrc.tgac.miso.core.data.impl.view.BoxableView;
-import uk.ac.bbsrc.tgac.miso.core.factory.barcode.BarcodeFactory;
-import uk.ac.bbsrc.tgac.miso.core.manager.MisoFilesManager;
 import uk.ac.bbsrc.tgac.miso.core.manager.RequestManager;
 import uk.ac.bbsrc.tgac.miso.core.service.naming.NamingScheme;
 import uk.ac.bbsrc.tgac.miso.core.service.naming.validation.ValidationResult;
 import uk.ac.bbsrc.tgac.miso.core.util.LimsUtils;
```
Comment (sqwci) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/SampleControllerHelperService.java]: MINOR Remove this unused import 'uk.ac.bbsrc.tgac.miso.core.util.LimsUtils'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/SampleControllerHelperService.java [hunk 4]
```diff
@@ -24,22 +24,15 @@
 package uk.ac.bbsrc.tgac.miso.spring.ajax;
 
 import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.*;
-import static uk.ac.bbsrc.tgac.miso.spring.ControllerHelperServiceUtils.getBarcodeFileLocation;
 
-import java.awt.image.RenderedImage;
-import java.io.File;
 import java.io.IOException;
 import java.util.ArrayList;
```
Comment (sqwci) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/SampleControllerHelperService.java]: MINOR Remove this unused import 'java.util.ArrayList'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/SampleControllerHelperService.java [hunk 5]
```diff
@@ -24,22 +24,15 @@
...
-import java.util.Arrays;
 import java.util.Date;
 import java.util.List;
```
Comment (sqwci) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/SampleControllerHelperService.java]: MINOR Remove this unused import 'java.util.List'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java [hunk 6]
```diff
@@ -54,15 +45,10 @@
...
 import net.sourceforge.fluxion.ajax.util.JSONUtils;
 
-import uk.ac.bbsrc.tgac.miso.core.data.Boxable;
 import uk.ac.bbsrc.tgac.miso.core.data.Library;
-import uk.ac.bbsrc.tgac.miso.core.data.impl.view.BoxableView;
-import uk.ac.bbsrc.tgac.miso.core.factory.barcode.BarcodeFactory;
-import uk.ac.bbsrc.tgac.miso.core.manager.MisoFilesManager;
 import uk.ac.bbsrc.tgac.miso.core.service.naming.NamingScheme;
 import uk.ac.bbsrc.tgac.miso.core.service.naming.validation.ValidationResult;
 import uk.ac.bbsrc.tgac.miso.core.util.LimsUtils;
```
Comment (sqwci) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java]: MINOR Remove this unused import 'uk.ac.bbsrc.tgac.miso.core.util.LimsUtils'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java [hunk 7]
```diff
@@ -24,23 +24,14 @@
...
 
 import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.*;
-import static uk.ac.bbsrc.tgac.miso.spring.ControllerHelperServiceUtils.getBarcodeFileLocation;
 
-import java.awt.image.RenderedImage;
-import java.io.File;
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Collection;
```
Comment (sqwci) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java]: MINOR Remove this unused import 'java.util.Collection'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java [hunk 8]
```diff
@@ -24,23 +24,14 @@
...
 import java.util.HashMap;
```
Comment (sqwci) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java]: MINOR Remove this unused import 'java.util.HashMap'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java [hunk 9]
```diff
@@ -24,23 +24,14 @@
...
-import java.util.List;
 import java.util.Map;
```
Comment (sqwci) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java]: MINOR Remove this unused import 'java.util.Map'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by sqwci** [comment]
SonarQube analysis reported 8 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 7 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HibernatePoolableElementViewDao.java#L27](https://github.com/TGAC/miso-lims/blob/13437a1990a5390d2ee449193f56169481b00b46/sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernatePoolableElementViewDao.java#L27): Define a constant instead of duplicating this literal "dilutionBarcode" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/page/AbstractListPage.java [hunk 10]
```diff
@@ -64,6 +64,7 @@
     public static final String COMMISSIONED = "Commissioned";
     public static final String DECOMMISSIONED = "Decommissioned";
     public static final String VOLUME = "Volume";
+    public static final String VOLUME = "Matrix Barcode";
```
Comment (callunity) [inline miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/page/AbstractListPage.java]: should be `MATRIX_BARCODE` not `VOLUME`

**Comment by sqwci** [comment]
SonarQube analysis reported 8 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 7 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HibernatePoolableElementViewDao.java#L27](https://github.com/TGAC/miso-lims/blob/ca402d71acc85bb394e148f0ab87ed1737b52439/sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernatePoolableElementViewDao.java#L27): Define a constant instead of duplicating this literal "dilutionBarcode" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HibernatePoolableElementViewDao.java#L27](https://github.com/TGAC/miso-lims/blob/8d9ff3f69284bd4b4bd0b7a35e9e605b547d1112/sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernatePoolableElementViewDao.java#L27): Define a constant instead of duplicating this literal "dilutionBarcode" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Pull request merged**